src/sage/graphs/domination.py

```diff
@@ -1284,3 +1285,45 @@ def greedy_dominating_set(G, k=1, vertices=None, ordering=None, return_sets=Fals
         return dom
     else:
         return list(dom)
+
+
+# ==============================================================================
+# Maximum leaf number
+# ==============================================================================
+
+def max_leaf_number(G):
+    r"""
+    Return the maximum leaf number of the graph, i.e. the maximum possible number
```

Comment:
Instead of "the complement of the caridnality", this needs to be "the cardinality of the complement". (In addition to changing the order, notice that there is a typo in "caridnality".)